core/trino-main/src/main/java/io/trino/operator/table/Sequence.java

```diff
@@ -0,0 +1,362 @@
+import io.trino.plugin.base.classloader.ClassLoaderSafeConnectorTableFunction;
+import io.trino.spi.HostAddress;
+import io.trino.spi.Page;
+import io.trino.spi.TrinoException;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorSplit;
+import io.trino.spi.connector.ConnectorSplitSource;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.connector.FixedSplitSource;
+import io.trino.spi.ptf.AbstractConnectorTableFunction;
+import io.trino.spi.ptf.Argument;
+import io.trino.spi.ptf.ConnectorTableFunction;
+import io.trino.spi.ptf.ConnectorTableFunctionHandle;
+import io.trino.spi.ptf.ReturnTypeSpecification.DescribedTable;
+import io.trino.spi.ptf.ScalarArgument;
+import io.trino.spi.ptf.ScalarArgumentSpecification;
+import io.trino.spi.ptf.TableFunctionAnalysis;
+import io.trino.spi.ptf.TableFunctionProcessorProvider;
+import io.trino.spi.ptf.TableFunctionProcessorState;
+import io.trino.spi.ptf.TableFunctionSplitProcessor;
+
+import javax.inject.Provider;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.airlift.slice.SizeOf.instanceSize;
+import static io.trino.metadata.GlobalFunctionCatalog.BUILTIN_SCHEMA;
+import static io.trino.operator.table.Sequence.SequenceFunctionSplit.DEFAULT_SPLIT_SIZE;
+import static io.trino.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+import static io.trino.spi.ptf.Descriptor.descriptor;
+import static io.trino.spi.ptf.TableFunctionProcessorState.Finished.FINISHED;
+import static io.trino.spi.ptf.TableFunctionProcessorState.Processed.produced;
+import static io.trino.spi.ptf.TableFunctionProcessorState.Processed.usedInputAndProduced;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static java.lang.Math.addExact;
+import static java.lang.Math.multiplyExact;
+import static java.lang.String.format;
+
+public class Sequence
+        implements Provider<ConnectorTableFunction>
+{
+    public static final String NAME = "sequence";
+
+    @Override
+    public ConnectorTableFunction get()
+    {
+        return new ClassLoaderSafeConnectorTableFunction(new SequenceFunction(), getClass().getClassLoader());
+    }
+
+    public static class SequenceFunction
+            extends AbstractConnectorTableFunction
+    {
+        private static final String START_ARGUMENT_NAME = "START";
+        private static final String STOP_ARGUMENT_NAME = "STOP";
+        private static final String STEP_ARGUMENT_NAME = "STEP";
+
+        public SequenceFunction()
+        {
+            super(
+                    BUILTIN_SCHEMA,
+                    NAME,
+                    ImmutableList.of(
+                            ScalarArgumentSpecification.builder()
+                                    .name(START_ARGUMENT_NAME)
+                                    .type(BIGINT)
```

Comment:
Do PTFs support overloading? Would it be possible to later add a sequence function for dates and timestamps?

Reply:
Table functions don't support overloading other than resulting from automatic type coercion. For dates or timestamps we'd have to add separate functions. Alternatively, we could model overloading by specifying optional arguments.

Reply:
That's in line with what I was thinking. I chose the name `sequence` for the most popular sequence function, and considered adding `date_sequence` as a follow-up.

Overloading with optional arguments is just a theoretical possibility. I wouldn't do it unless the overloads share most arguments.